src/Bridges/Constraint/bridges/functionize.jl

```diff
@@ -322,3 +322,111 @@ function MOI.get(
     f = MOI.get(model, attr, b.constraint)
     return MOI.Utilities.convert_approx(MOI.VectorOfVariables, f)
 end
+
+"""
+    ScalarQuadraticToScalarNonlinearBridge{T,S} <: Bridges.Constraint.AbstractBridge
+
+`ScalarQuadraticToScalarNonlinearBridge` implements the following reformulations:
+
+  * ``f(x) \\in S`` into ``g(x) \\in S``
+
+where `f` is a [`MOI.ScalarQuadraticFunction`](@ref) and `g` is a
+[`MOI.ScalarNonlinearFunction{T}`](@ref).
+
+## Source node
+
+`ScalarQuadraticToScalarNonlinearBridge` supports:
+
+  * [`MOI.ScalarQuadraticFunction`](@ref) in `S`
+
+## Target nodes
+
+`ScalarQuadraticToScalarNonlinearBridge` creates:
+
+  * [`MOI.ScalarNonlinearFunction{T}`](@ref) in `S`
+"""
+struct ScalarQuadraticToScalarNonlinearBridge{T,S} <:
+       AbstractFunctionConversionBridge{MOI.ScalarNonlinearFunction,S}
+    constraint::MOI.ConstraintIndex{MOI.ScalarNonlinearFunction,S}
+end
+
+const ScalarQuadraticToScalarNonlinear{T,OT<:MOI.ModelLike} =
+    SingleBridgeOptimizer{ScalarQuadraticToScalarNonlinearBridge{T},OT}
+
+function bridge_constraint(
+    ::Type{ScalarQuadraticToScalarNonlinearBridge{T,S}},
+    model::MOI.ModelLike,
+    f::MOI.ScalarQuadraticFunction{T},
+    s::S,
+) where {T,S}
+    ci = MOI.add_constraint(model, convert(MOI.ScalarNonlinearFunction, f), s)
+    return ScalarQuadraticToScalarNonlinearBridge{T,S}(ci)
+end
+
+function MOI.supports_constraint(
+    ::Type{ScalarQuadraticToScalarNonlinearBridge{T}},
+    ::Type{MOI.ScalarQuadraticFunction{T}},
+    ::Type{<:MOI.AbstractScalarSet},
+) where {T}
+    return true
+end
+
+function MOI.Bridges.added_constrained_variable_types(
+    ::Type{<:ScalarQuadraticToScalarNonlinearBridge},
+)
+    return Tuple{Type}[]
+end
+
+function MOI.Bridges.added_constraint_types(
+    ::Type{ScalarQuadraticToScalarNonlinearBridge{T,S}},
+) where {T,S}
+    return Tuple{Type,Type}[(MOI.ScalarNonlinearFunction, S)]
+end
+
+function concrete_bridge_type(
+    ::Type{<:ScalarQuadraticToScalarNonlinearBridge{T}},
+    ::Type{MOI.ScalarQuadraticFunction{T}},
+    S::Type{<:MOI.AbstractScalarSet},
+) where {T}
+    return ScalarQuadraticToScalarNonlinearBridge{T,S}
+end
+
+function MOI.get(
```

Comment:
This method and the next two should be implemented at the `AbstractFunctionConversion` level

Reply:
But this can be done in a separate PR

Reply:
Yeah there's the canonical issue as well. But we can come back to it with https://github.com/jump-dev/MathOptInterface.jl/pull/2224